### PR TITLE
Don't offer to add a null check for an unknown type when one already exists.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/19173

**Customer scenario**

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

**Bugs this fixes:**

(either VSO or GitHub links)

**Workarounds, if any**

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

**Risk**

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

**Performance impact**

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

**Is this a regression from a previous update?**

**Root cause analysis:**

How did we miss it?  What tests are we adding to guard against it in the future?

**How was the bug found?**

(E.g. customer reported it vs. ad hoc testing)

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -117,7 +117,10 @@ private bool IsIfNullCheck(IOperation statement, IParameterSymbol parameter)
         {
             if (statement is IIfStatement ifStatement)
             {
-                if (ifStatement.Condition is IBinaryOperatorExpression binaryOperator)
+                var condition = ifStatement.Condition;
+                condition = UnwrapImplicitConversion(condition);
```
Comment (Pilchie) [inline src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs]: How does this fix the issue?

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs]: The compiler wraps error types in an implicit conversion to 'object'.  So the check of "args == null" ends up not being seen as a null check.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 


**Customer scenario**

This is a fairly simple fix that prevents the user from continually getting the same lightbulb fix from us over and over again, even once they've accepted the fix.

**Bugs this fixes:**

Fixes https://github.com/dotnet/roslyn/issues/19173

**Workarounds, if any**

None

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.  New feature.

**Root cause analysis:**

Insufficient testing over possible user error conditions.

**How was the bug found?**

Customer reported.

**Pull request merged**